ccmlib/node.py

```diff
@@ -772,7 +773,10 @@ def matches(expected, actual):
 
         total = 0
         for line in tail:
-            ks, cf, num = tail_pattern.search(line).groups()
+            m = tail_pattern.search(line)
+            if not m:
```

Comment:
if m is None:


More specific this way.

Reply:
Why do we care?
`not m` is simpler and sensible

Reply:
i don't feel strong either way. It's like `m == nullptr` versus `!m` in c/c++ world. The latter works as well, but one cannnot deduce that `m` is a pointer by reading the code.

Reply:
Right, and I prefer the boolean operator rather than comparing to the null value where it's not needed explicitly, like when you to distinguish between an arg default value of None vs. a valid value of False or [].
In the latter case, None would mean unset, not just False.

Reply:
yeah, I don't insist in this case. But, personally , I prefer explicit even at the expense of more verbosity.